align/compiler/match_graph.py

```diff
@@ -165,7 +165,7 @@ def pin_id(pins):
         param = FlatDict({"_":all_properties})
         arg_str = '_'.join([k+':'+str(param[k]) for k in sorted(param.keys())])
         key = f"_{str(int(hashlib.sha256(arg_str.encode('utf-8')).hexdigest(), 16) % 10**8)}"
-        new_subckt_name = (const.template_name if const.template_name else 'primitive')+key
+        new_subckt_name = (const.template_name if const.template_name else 'primitive'+key)
```

Comment:
What is this change? This is causing test failure. Can we revert this change from this PR?

Reply:
This change is to find a subcircuit easily in the library (without relying on the hash code).
See line 198 in test_auto_const.py below.
If user is specifying a template (i.e., subcircuit) name, why append a hash to it?

Reply:
You are welcome to revert this change but please update the new tests as well.

Reply:
I will revert this for now and handle it in a separate PR.